### PR TITLE
Update snap and keyring api dependencies and fix tests

### PR DESCRIPTION
**Pull request by montelaidev**
This pull request updates the dependencies in the project, including bumping snap dependencies, keyring-api, and eth-snap-keyring. It also fixes the failing tests.

Closes: https://github.com/MetaMask/accounts-planning/issues/307

### CODE REVIEW

**Comment by montelaidev** [comment]
@SocketSecurity ignore npm/@metamask/post-message-stream@8.0.0
@SocketSecurity ignore npm/@metamask/eth-snap-keyring@3.0.0

File: packages/user-operation-controller/src/UserOperationController.ts [hunk 1]
```diff
@@ -532,6 +532,7 @@ export class UserOperationController extends BaseController<
 
     const response = await smartContractAccount.updateUserOperation({
       userOperation,
+      chainId: metadata.chainId,
```
Comment (matthewwalsh0) [inline packages/user-operation-controller/src/UserOperationController.ts:535]: Can we use destructuring to get this on line 529?

File: packages/user-operation-controller/src/helpers/SnapSmartContractAccount.ts [hunk 2]
```diff
@@ -92,6 +98,9 @@ export class SnapSmartContractAccount implements SmartContractAccount {
       'KeyringController:signUserOperation',
       sender,
       userOperation,
+      {
+        chainId: request.chainId,
```
Comment (matthewwalsh0) [inline packages/user-operation-controller/src/helpers/SnapSmartContractAccount.ts:102]: Same for these three, can we destructure them when we first access `request` at the start of the method?

**Review by matthewwalsh0 — APPROVED** [review]

**Comment by ccharly** [comment]
I'm gonna drop this [branch](https://github.com/MetaMask/core/tree/feature/307/caip-2-user-ops) in favor of this PR (I've backported everything here).

**Review by matthewwalsh0 — APPROVED** [review]

**Pull request merged**